feat(AppHeaderProfileDropdown): Pass through user avatar and display name

katiezutter opened the pull request:
### Overview

<!--- CHANGELOG-DESCRIPTION -->

Adds AppHeaderProfileDropdown component and sets up mobile menu for account sub-menu.

<!--- END-CHANGELOG-DESCRIPTION -->

This PR was initially meant to set up the App Header items that we needed for the account sub-menu for the mobile header:

<img width="506" alt="Screen Shot 2021-01-27 at 3 50 46 PM" src="https://user-images.githubusercontent.com/14988121/106058920-887b5d80-60b7-11eb-95c9-7cdfb5f32708.png">

While doing this, I realized that since we now have the avatar, the user's display name, and all of the links that we needed for the mobile account sub-menu, we might as well have the desktop profile dropdown on the Gamut side as well (`AppHeaderProfileDropdown`). You can see this in the preview storybook for `GlobalHeader`:

<img width="462" alt="Screen Shot 2021-01-27 at 3 58 11 PM" src="https://user-images.githubusercontent.com/14988121/106059541-78b04900-60b8-11eb-9a34-451eeb42cf99.png">

This PR also addresses a z-index issue where the profile dropdown content was appearing underneath other content on the page. To address this, we now have an optional parameter for the consumer of `GlobalHeader` to pass in a base z-index.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [EGG-663](https://codecademy.atlassian.net/browse/EGG-663)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

<!--
Merging your changes

1. Follow the [PR Title Guide](https://github.com/Codecademy/client-modules#pr-title-guide), the title (which becomes the commit message) determines the version bump for the packages you changed.

2. Wrap the text describing your change in more detail in the "CHANGELOG-DESCRIPTION" comment tags above, this is what will show up in the changelog!

3. DO NOT MERGE MANUALLY! When you are ready to merge and publish your changes, add the "Ship It" label to your Pull Request. This will trigger the merge process as long as all checks have completed, if the checks haven't completed the branch will be merged when they all pass.

**IMPORTANT:** If your PR contains breaking changes, please remember to follow the instructions for breaking changes!
-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.16.1-alpha.183f34.0<br/>@codecademy/styleguide@21.24.1-alpha.183f34.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6011e401d8fb091058a1bb98--gamut-preview.netlify.app>https://6011e401d8fb091058a1bb98--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6011e401d8fb091058a1bb98>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

saghdaey commented on packages/gamut-labs/src/experimental/AppHeaderMobile/AppHeaderSubMenuMobile/index.tsx:
```diff
@@ -4,22 +4,28 @@ import { pxRem } from '@codecademy/gamut-styles';
-export type AppHeaderSubMenuMobileProps = AppHeaderDropdownProps & {
+export type AppHeaderSubMenuMobileProps = (
+  | AppHeaderDropdownProps
+  | AppHeaderProfileDropdownProps
```
nice

saghdaey commented on packages/gamut-labs/src/experimental/AppHeaderMobile/AppHeaderLinkMobile/index.tsx:
```diff
@@ -6,14 +6,32 @@ import {
 
 export type AppHeaderLinkMobileProps = {
-  onClick: (event: React.MouseEvent) => {};
+  arrowButton?: boolean;
```
did you just add this in preparation for this being able to open a submenu? I don't see arrowButton being used in the component (yet?)

katiezutter replied:
Oh I didn't add this in, it must've been from your other branch that I started with! I can take it out if you want, but maybe it's fine if you'll need it for your next PR?

saghdaey replied:
oh haha yep must be thats funny:) 
i can handle it np!!

katiezutter commented on packages/gamut-labs/src/experimental/AppHeaderMobile/AppHeaderLinkMobile/index.tsx:
```diff
@@ -30,28 +48,33 @@ export const AppHeaderLinkMobile: React.FC<AppHeaderLinkMobileProps> = ({
+          </AppHeaderLinkButtonInner>
+        </AppHeaderLinkButtonOuter>
+      </SeparatorInner>
+    </SeparatorOuter>
```
This whole deal of having a bunch of styled components here isn't my favorite, so suggestions are welcome! This is the way I could figure it out so that the styling for the horizontal separators looked right.

saghdaey commented:
I think we may need to add an admin variant to the pro dropdown to handle these other options --> 
<img width="398" alt="Screen Shot 2021-01-27 at 5 20 08 PM" src="https://user-images.githubusercontent.com/9753446/106061823-ed3cb500-60c3-11eb-9dc6-9ced75319c69.png">

JoshuaKGoldberg commented on packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderProfileDropdown.stories.mdx:
```diff
@@ -0,0 +1,72 @@
+  component={AppHeaderProfileDropdown}
+  parameters={{ layout: 'padded', status: 'volatile', source: 'gamut-labs' }}
+  args={{
+    item: {
```
Oof, https://github.com/storybookjs/storybook/pull/12824 would be really nice here...
![Screenshot of a bunch of raw text for a JSON 'item' prop in Storybook](https://user-images.githubusercontent.com/3335181/106092045-90f58780-60fb-11eb-82db-2979f593e99f.png)

JoshuaKGoldberg commented on packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderProfileDropdown.stories.mdx:
```diff
@@ -0,0 +1,72 @@
+
+<Meta
+  title={title}
+  component={AppHeaderProfileDropdown}
```
Looks like there's a CSS bug with this in at least the story?

![Screenshot of the profile dropdown floating weirdly to the right](https://user-images.githubusercontent.com/3335181/106092301-ff3a4a00-60fb-11eb-9158-09f8ff58a052.png)

katiezutter replied:
Ahh yes, noticed this and then forgot to circle back! I think it's a storybook thing, looking into it.

JoshuaKGoldberg commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderProfileDropdown/__tests__/AppHeaderProfileDropdown-test.tsx:
```diff
@@ -0,0 +1,89 @@
+        href: testDropdownUrls[2],
+        topSeparator: true,
+        trackingTarget: '',
+        type: 'link',
```
You might consider making a `createMockAppHeaderLinkItem` function to reduce duplication in all of these 😉 

![Screenshot of four test files with popover data](https://user-images.githubusercontent.com/3335181/106092434-4aecf380-60fc-11eb-805c-897614fbc552.png)

JoshuaKGoldberg commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderProfileDropdown/__tests__/AppHeaderProfileDropdown-test.tsx:
```diff
@@ -0,0 +1,89 @@
+  it('sublinks link to the provided hrefs', () => {
+    renderAppHeaderProfileDropdown();
+    screen.getByRole('button').click();
+    expect(screen.getAllByRole('link')[0].getAttribute('href')).toEqual(
```
Up for using a `expect(....map(node => node.getAttribute('href'))).toEqual(...)` to make this all into one assertion?

katiezutter replied:
Yes!! Thanks for these great improvements

JoshuaKGoldberg reviewed (changes requested):
This is looking visually beautiful, mostly non-blocking nits! Just the one blocker IMO from the story having the dropdown float off to the right.

JoshuaKGoldberg commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderProfileDropdown/index.tsx:
```diff
@@ -0,0 +1,79 @@
+        baseZIndex={baseZIndex}
+        verticalOffset={0}
+        outline
+        isOpen={isOpen}
```
Hmm, this code looks really similar to `AppHeaderDropdown`... is there some DRYing that should happen?

katiezutter replied:
That's the path I was originally going down, but then once the file started looking convoluted I split it up - I'll take another stab at it!

katiezutter replied:
Update: oh wow this was _much_ easier now after the types `profile-dropdown` and `dropdown` are in order!

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.16.1-alpha.06724e.0<br/>@codecademy/styleguide@21.24.1-alpha.06724e.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6012eac023448f35df30cc64--gamut-preview.netlify.app>https://6012eac023448f35df30cc64--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6012eac023448f35df30cc64>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

katiezutter commented:
@saghdaey there is that admin functionality in this PR!

you can see it here for admin: https://6012eac023448f35df30cc64--gamut-preview.netlify.app/storybook/?path=/docs/labs-experimental-molecules-globalheader--pro-user-global-header-admin

and here for CS: https://6012eac023448f35df30cc64--gamut-preview.netlify.app/storybook/?path=/docs/labs-experimental-molecules-globalheader--pro-user-global-header-customer-support

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601312411768fe4ff35925f0--gamut-preview.netlify.app>https://601312411768fe4ff35925f0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601312411768fe4ff35925f0>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

katiezutter commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderProfileDropdown/index.tsx:
```diff
@@ -0,0 +1,79 @@
+  const clickTarget = (
+    <AppHeaderAvatarTargetButton
+      onClick={(event) => toggleIsOpen(event)}
+      style={{ paddingTop: pxRem(2), paddingBottom: pxRem(2) }}
```
No good reason - changed!

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.16.5-alpha.f88d81.0<br/>@codecademy/styleguide@22.0.4-alpha.f88d81.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60132f6fce29f3809c6b3208--gamut-preview.netlify.app>https://60132f6fce29f3809c6b3208--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60132f6fce29f3809c6b3208>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

katiezutter commented:
I addressed all of the feedback - thanks y'all! 😊 @JoshuaKGoldberg @saghdaey I'm re-requesting review because I made some pretty major refactors to this work! Also, thanks goodness commits will be squashed...yikes.

JoshuaKGoldberg commented on packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderDropdown.stories.mdx:
```diff
@@ -101,6 +102,21 @@ A dropdown menu for use in AppHeader.
   </Story>
 </Canvas>
+
+<!-- These breaks are to allow room on the page for the Profile dropdown above -->
```
You could do this a teeny bit more elegantly with some `min-height` and/or `margin-bottom` in CSS 😉

dreamwasp commented on packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderDropdown.stories.mdx:
```diff
@@ -41,3 +43,80 @@ A dropdown menu for use in AppHeader.
+    name="AppHeaderProfileDropdown"
+  >
+    {(args) => (
+      <Box height="50">
```
maybe you can add `margin`'s to this `Box` instead of the `<br />`s below?

dreamwasp commented on packages/gamut-labs/src/experimental/AppHeader/index.tsx:
```diff
@@ -82,17 +92,18 @@ const mapItemToElement = (
     <AppBar className={className}>
       <AppBarSection position="left">
-        {items.left.map((item) => mapItemToElement(item, onClick))}
+        {items.left.map((item) => mapItemToElement(baseZIndex, item, onClick))}
```
just out of curiosity, when you one need to pass an onClick that applied to the entire Header? tracking?

saghdaey replied:
yep! tracking 
& and then hopefully in the future we can add routing via redux-first-router as well

dreamwasp reviewed (commented):
LGTM, just have a couple questions above + am curious about the preexisting conditions that led to needing to define the z-index in this way

saghdaey commented on packages/gamut-labs/src/experimental/AppHeaderMobile/AppHeaderSubMenuMobile/index.tsx:
```diff
@@ -4,22 +4,22 @@ import { pxRem } from '@codecademy/gamut-styles';
 
-import { AppHeaderDropdownProps } from '../../..';
 import { Text } from '../../../experimental/Text';
-import { AppHeaderLink } from '../../AppHeader/AppHeaderElements/AppHeaderLink';
```
I had intentionally used the AppHeaderLink component here instead of the AppHeaderLinkMobile, is there a reason you switched it? 
My thought was the former never shows an icon (which is the behavior we want in the AppHeaderSubMenuMobile why use the component (AppHeaderLinkMobile) that will always check for something that's false?

katiezutter replied:
It's because we'll be reusing this component for the Profile sub-menu which does have icons, unlike the other sub-menus!

saghdaey replied:
oooo nice nice forgot that one had icons

saghdaey commented on packages/gamut-labs/src/experimental/GlobalHeader/GlobalHeaderItems.tsx:
```diff
@@ -171,13 +180,124 @@ export const notifications = (
+const profileAdmin: AppHeaderLinkItem = {
+  id: 'admin',
+  href: '/admin',
+  topSeparator: true,
```
I don't love that we're adding specific styling properties to the AppHeaderItems; I wonder if there's a way of around this that's still easy to read

katiezutter replied:
Another option would be to do an array of an array for profile dropdown items, so for an admin the profile dropdown would be something like [[profile, account, etc], [admin, cs, reportBug], [logout]] and then the dropdown could add a top separator to the first element of each array (except the first) - do you think that would be the better way to go?

JoshuaKGoldberg commented on packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:
```diff
@@ -32,40 +44,106 @@ The following are available variations for the GlobalHeader component.
 <Canvas>
   <Story name="Anonymous User Global Header">
-    {(args) => <GlobalHeader type="anon" />}
+    {(args) => <GlobalHeader onClick={() => {}} type="anon" />}
```
You might consider using `action('...')` to be a little more informative than `onClick` for these handlers. Search `action('` to see other examples.

katiezutter commented on packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:
```diff
@@ -32,40 +44,106 @@ The following are available variations for the GlobalHeader component.
 </Canvas>
 
-### Anonymous User - Default
+### Anonymous User
```
Yeah, can't figure out why that's happening in the Storybook. @saghdaey and I spent a deal of time trying to figure out how to make it appear better on the Docs page but didn't figure it out! If you toggle to Canvas instead of Docs in Storybook after selecting each variant in the left-hand sidebar, that's more what it looks like 😊

JoshuaKGoldberg replied:
We'll probably need to add some kind of `<Meta ... />` prop to specify that this page needs wide styles. I think the page is just not wide enough for it.

![Screenshot of not very wide page shrinking the header width](https://user-images.githubusercontent.com/3335181/106504980-ae2cac00-6495-11eb-85a1-2e0f5a57baea.png)

fyi @codecaaron

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.16.5-alpha.b7f370.0<br/>@codecademy/styleguide@22.0.4-alpha.b7f370.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60143deb237e6e66885b982b--gamut-preview.netlify.app>https://60143deb237e6e66885b982b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60143deb237e6e66885b982b>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

katiezutter commented:
@dreamwasp glad you asked about the z-index thing - we've been through a little bit of an adventure with that, and just paired with Aaron earlier this afternoon! Took it out of this PR and we talked through some changes within static-sites/monolith that need to happen when we switch over to the new components! So now there are no z-index changes here. It was looking funky in Storybook though so we made Canvases have a height of 500px in the GlobalHeader story.

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@8.0.3-alpha.114ea8.0<br/>@codecademy/styleguide@23.1.1-alpha.114ea8.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601488f6fff37452dde582db--gamut-preview.netlify.app>https://601488f6fff37452dde582db--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601488f6fff37452dde582db>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

katiezutter commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLinkSections/index.tsx:
```diff
@@ -0,0 +1,58 @@
+          })}
+    </>
+  );
+};
```
@saghdaey this is the component I added after revamping the way we're passing in link sections to the profile dropdown!

katiezutter commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/types.tsx:
```diff
@@ -51,6 +53,15 @@ export type AppHeaderDropdownItem = AppHeaderBaseItem & {
+export type AppHeaderProfileDropdownItem = AppHeaderBaseItem & {
+  avatar: string;
+  displayName: string;
+  popover: AppHeaderLinkItem[][];
```
@saghdaey and here I changed it to a 2D array for the links, vs that `topSeparator` stuff

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@8.0.3-alpha.46a47e.0<br/>@codecademy/styleguide@23.1.1-alpha.46a47e.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60148ced34cec90090e08bbe--gamut-preview.netlify.app>https://60148ced34cec90090e08bbe--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60148ced34cec90090e08bbe>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

saghdaey commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderAvatar/index.tsx:
```diff
@@ -0,0 +1,33 @@
+`;
+
+export type AppHeaderAvatarProps = {
+  imageUrl?: string;
```
do we have a default imageUrl? why is this optional?

katiezutter replied:
Good catch - shouldn't be!

saghdaey commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderAvatar/index.tsx:
```diff
@@ -0,0 +1,33 @@
+  border-radius: 100%;
+  width: 2.5rem;
+  height: 2.5rem;
+`;
```
maybe we can use box but non-blocking!

saghdaey commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/__tests__/AppHeaderDropdown-test.tsx:
```diff
@@ -56,11 +51,8 @@ describe('AppHeaderDropdown', () => {
-    );
+    expect(
+      screen.getAllByRole('link').map((node) => node.getAttribute('href'))
+    ).toStrictEqual(testDropdownUrls);
```
nice:)

saghdaey commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:
```diff
@@ -32,60 +32,76 @@ const AppHeaderTargetButton = styled.button`
+        className={isOpen && styles.open}
+        onClick={(event) => toggleIsOpen(event)}
+      >
+        <span title={item.text} className={styles.copy}>
```
how come this span has a title? I think I had this in one of my PR's and josh or aaron asked me... and I didnt know and deleted lol

katiezutter replied:
Really I'm just trying to secretly work back in all the unnecessary code that you deleted haha

saghdaey commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLinkSections/index.tsx:
```diff
@@ -0,0 +1,58 @@
+      {item.type === 'profile-dropdown'
+        ? item.popover.map((linkSection: AppHeaderLinkItem[], sectionIndex) => {
+            return linkSection.map((link: AppHeaderLinkItem, linkIndex) => {
+              if (sectionIndex !== 0 && linkIndex === 0) {
+                return (
+                  <Box key={link.id} paddingX={16}>
+                    <AppHeaderLinkMobile
+                      action={action}
+                      item={link}
+                      key={link.id}
+                      topSeparator={true}
+                    />
+                  </Box>
+                );
+              }
+              return (
+                <Box key={link.id} paddingX={16}>
+                  <AppHeaderLinkMobile
+                    action={action}
+                    item={link}
+                    key={link.id}
+                  />
+                </Box>
```
is the only difference between these two returns, the topSeparator true?  That's all i'm seeing but I dont trust my eyes rn. 
 I wonder if we could make this a little shorter and do --
`
                return (
                  <Box key={link.id} paddingX={16}>
                    <AppHeaderLinkMobile
                      action={action}
                      item={link}
                      key={link.id}
                      topSeparator={showTopSeparator(sectionIndex, linkIndex)}
                    />
                  </Box>

`

saghdaey replied:
ok sorry that formatting did not work lol

saghdaey replied:
& we can define showTopSeparator outside of the return

katiezutter replied:
Duh! Thanks for catching this!

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@8.0.3-alpha.27313d.0<br/>@codecademy/styleguide@23.1.1-alpha.27313d.0<br/></pre><!--LERNA_VERSION_COMMENT-->

saghdaey commented on packages/gamut-labs/src/experimental/AppHeader/mockAppHeaderItems.tsx:
```diff
@@ -0,0 +1,16 @@
+  href: string,
+  text: string
+): AppHeaderLinkItem => {
+  return {
```
nice love this v useful

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6014980413fd1710b53ef028--gamut-preview.netlify.app>https://6014980413fd1710b53ef028--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6014980413fd1710b53ef028>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@8.0.3-alpha.7bb6ae.0<br/>@codecademy/styleguide@23.1.1-alpha.7bb6ae.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601499c3d3430b1010c875bf--gamut-preview.netlify.app>https://601499c3d3430b1010c875bf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601499c3d3430b1010c875bf>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

JoshuaKGoldberg reviewed (approved):
Awesome! Left a little bit of DRY nitting, but nothing blocking. 🌮 

I suspect that this header will need some tweaking once we get it into the monolith and static-sites, but that's normal. Let's get it in!

katiezutter commented on packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLinkSections/index.tsx:
```diff
@@ -0,0 +1,48 @@
+            return linkSection.map((link: AppHeaderLinkItem, linkIndex) => {
+              const showTopSeparator = sectionIndex !== 0 && linkIndex === 0;
+              return (
+                <Box key={link.id} paddingX={16}>
```
If `type === 'profile-dropdown'` then we do the double mapping, first mapping over each section (`AppHeaderLinkItem[]`) then each `AppHeaderLinkItem`, vs. a normal dropdown, which does not have a 2D array of links, just maps over `AppHeaderLinkItem`s. But I was able to change the `Box` and key things!

JoshuaKGoldberg replied:
Oh gotcha - thanks!

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@8.0.3-alpha.e25339.0<br/>@codecademy/styleguide@23.1.2-alpha.e25339.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60185b99d1b09419c0c3363a--gamut-preview.netlify.app>https://60185b99d1b09419c0c3363a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60185b99d1b09419c0c3363a>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->
